_studio/shared/umc/codec/vc1_dec/include/umc_vc1_dec_task_store.h

```diff
@@ -182,17 +182,17 @@ namespace UMC
         };
 
         // external memory management. No need to delete memory
-        void operator delete(void *p) THROWSEXCEPTION
+        void operator delete(void *p)
         {
             //Anyway its incorrect when we trying free null pointer
-            if (!p)
-                throw VC1Exceptions::vc1_exception(VC1Exceptions::mem_allocation_er);
+            VM_ASSERT(p);
+            (void) p;
         };
 
-        void operator delete(void *, void *) THROWSEXCEPTION
+        void operator delete(void *, void *)
         {
             // delete for system exceptions case
-            throw VC1Exceptions::vc1_exception(VC1Exceptions::mem_allocation_er);
+            VM_ASSERT(0);
```

Comment:
I also like this operator: looks like we have an operator which should _never_ be called:). Can't that be handled on compilation level via compile errors?

Reply:
In this case we should declare and not define, like this:
```
// By intent we prohibit this type of delete
void operator delete(void *, void *);
```

In that case we will get linker level error and avoid catching this at runtime.